server/routes/api/disaster_api.py

```diff
@@ -49,21 +50,134 @@ def event_date_range():
+    return True
+  prop_val = event[filter_prop]
+  if len(prop_val) < len(filter_unit):
+    # value for the property to filter by is empty
+    return True
+  val_string = prop_val[len(filter_unit):].strip()
+  try:
+    val = float(val_string)
+    return val <= filter_upper_limit and val >= filter_lower_limit
+  except:
+    # can't parse the value to filter by into a float
+    return True
+
+
+@bp.route('/json-event-data')
+def json_event_data():
+  """Gets the event data from saved jsons for a given eventType, date, place,
+     andfilter information (filter prop, unit, lower limit, and upper limit).
+
+  Returns: an object of the following form
+    {
+      eventCollection: {
+        events: [
+          {
+            dcid: string,
+            dates: list of strings,
+            geoLocations: list of { point: { latitude: number, longitude: number } },
+            places: list of strings,
+            provenanceId: string,
+            propVals: {
+              [prop]: { vals: list of strings },
+              ...
+            }
+          },
+          ...
+        ],
+        provenanceInfo: {
+          [provenanceId]: {
+            domain: string,
+            importName: string,
+            provenanceUrl: string
+          },
+          ...
+        }
+      }
+  """
+  event_type = request.args.get('eventType', '')
+  if not event_type:
+    return "error: must provide a eventType field", 400
+  date = request.args.get('date', '')
+  if not date:
+    return "error: must provide a date field", 400
+  place = request.args.get('place', '')
+  if not place:
+    return "error: must provide a place field", 400
+  filter_prop = request.args.get('filterProp', '')
+  filter_unit = request.args.get('filterUnit', '')
+  filter_upper_limit = float(request.args.get('filterUpperLimit', '0'))
+  filter_lower_limit = float(request.args.get('filterLowerLimit', '0'))
+  event_points = []
+  disaster_data = current_app.config['DISASTER_DASHBOARD_DATA']
+  for event in disaster_data.get(event_type, {}).get(date, []):
+    if (keep_event(event, place, filter_prop, filter_unit, filter_upper_limit,
+                   filter_lower_limit)):
+      event_formatted = {
+          "dcid": event["eventId"],
+          "dates": [event["startDate"]],
+          "places": event["affectedPlaces"],
+          "geoLocations": [{
+              "point": {
+                  "latitude": event["latitude"],
+                  "longitude": event["longitude"]
+              }
+          }],
+          "provenanceId": "",
+          "propVals": {}
+      }
+      for eventKey in event.keys():
+        if eventKey in EVENT_POINT_KEYS:
+          continue
+        event_formatted["propVals"][eventKey] = {"vals": [event[eventKey]]}
+      event_points.append(event_formatted)
+  result = {}
+  if event_points:
+    result = {"eventCollection": {"events": event_points, "provenanceInfo": {}}}
+  return json.dumps(result), 200
```

Comment:
suggest returning this as Response(json.dumps(result), 200, "application/json") so you can format these in the inspector.

Reply:
done